x-pack/legacy/plugins/ml/server/models/annotation_service/annotation.ts

```diff
@@ -67,7 +68,8 @@ export type callWithRequestType = (
   params: annotationProviderParams
 ) => Promise<any>;
 
-export function annotationProvider(callWithRequest: callWithRequestType) {
+export function annotationProvider(context: RequestHandlerContext) {
+  const callAsCurrentUser = context.ml!.mlClient.callAsCurrentUser;
```

Comment:
Can we work around this in another way? The styleguide says to avoid non-null assertions https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#avoid-non-null-assertions

Reply:
I agree, this has been bothering me too in the recent NP work.
why does `ml` need to be optional in our `RequestHandlerContext`? especially as we're asserting that it isn't optional in every route.

Reply:
Adding `ml` to the `RequestHandlerContext` interface was initially made optional to prevent issues with other plugins using `RequestHandlerContext`. This was the suggested definition from kibana-platform.
Had a chat with Josh about this and looks like they'll be improving this by adding a generic type arg to `core.http.createRouter` so you can specify your own type for `RequestHandlerContext`. The PR should be up soon. Then we'll be able to get rid of the non-null assertions. 🙌 
So we're stuck with this for now but once that PR is in we can add the change in the next routes PR 👌